pkg/networkserver/observability.go

```diff
@@ -274,7 +274,15 @@ var nsMetrics = &messageMetrics{
 			Help:      "Number of gateways that forwarded the uplink (within the deduplication window)",
 			Buckets:   []float64{1, 2, 3, 4, 5, 10, 20, 30, 40, 50},
 		},
-		[]string{},
+		nil,
+	),
+	micMismatches: metrics.NewContextualCounterVec(
```

Comment:
Considering that the total number of MIC checks is this number plus one, I would prefer just counting the total number of MIC checks we do, so that we can accurately calculate the average number of MIC checks per uplink.

Reply:
That's not entirely true with #2837 in place for LW1.1+
NS may perform 2 MIC checks - 1 only by CMACF, 1 complete MIC check (which later will reuse the CMACF value)
But registering a metric, which is triggered every time NS computes a MIC is fine with me. There will not be a clear way to determine, how many of them were unnecessary though.
Perhaps it makes more sense to have 2 separate metrics?

Reply:
See https://github.com/TheThingsNetwork/lorawan-stack/pull/3175/commits/fe499d78becd70c76179dcc6d1e76333b4c34474